analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -108,11 +126,27 @@ public bool HasMatchingScope(DiagnosticDescriptor descriptor)
             descriptor.CustomTags.Contains(tag);
     }
 
+    private bool ShouldAnalyzeFile(SonarLintXmlReader sonarLintXml, string filePath) =>
+        ProjectConfiguration().ProjectType != ProjectType.Unknown // Not SonarLint context, NuGet or Scanner <= 5.0
+        || (FileInclusionCache.GetValue(Compilation, _ => new()) is var cache
+            && cache.GetOrAdd(filePath, _ => IsFileIncluded(sonarLintXml, filePath)));
```

Comment:
The rename makes it way harder to read
```suggestion
        || (FileInclusionCache.GetValue(Compilation, _ => new()).GetOrAdd(filePath, x => IsFileIncluded(sonarLintXml, x)));
```

Reply:
To be addressed here https://github.com/SonarSource/sonar-dotnet/pull/6924